xla/backends/gpu/runtime/nvshmem_collective_thunk.h

```diff
@@ -125,6 +125,30 @@ absl::Status IsValidNvshmemOperand(Shape shape, Thunk::Kind reduction_op);
 
 absl::StatusOr<xla::gpu::GpuCollectives*> GetNvshmemCollectivesFromRegistry();
 
+// NvshmemBufferAddresses provides a mechanism to store and retrieve buffer
+// addresses for NVSHMEM put/get operations. This is necessary because NVSHMEM
+// uses one-way put/get operations where both source and destination addresses
+// are required.
+class NvshmemBufferAddresses {
+ public:
+  // Get buffer address for a device
+  absl::StatusOr<void*> GetNvshmemPtr(int device_ordinal);
+
+  // Store buffer address for a device
+  void StoreNvshmemPtr(int device_ordinal, void* buffer_addr);
+
+  // Get the global instance
+  static NvshmemBufferAddresses& GlobalNvshmemBufferAddresses() {
```

Comment:
Since this will be called as `NvshmemBufferAddresses::GlobalNvshmemBufferAddresses()` the name seems a bit repetitive. How about `NvshmemBufferAddresses::Global` or `NvshmemBufferAddresses::GlobalInstance`?